src/sqlfluff/core/linter/linting_result.py

```diff
@@ -146,6 +147,50 @@ def timing_summary(self) -> Dict[str, Dict[str, float]]:
                 timing.add(file.time_dict)
         return timing.summary()
 
+    def persist_timing_records(self, filename):
+        """Persist the timing records as a csv to external analysis."""
+        meta_fields = [
+            "path",
+            "source_chars",
+            "templated_chars",
+            "segments",
+            "raw_segments",
+        ]
+        timing_fields = ["templating", "lexing", "parsing", "linting"]
+        with open(filename, "w", newline="") as f:
+            writer = csv.DictWriter(f, fieldnames=meta_fields + timing_fields)
+
+            writer.writeheader()
+
+            for dir in self.paths:
+                for file in dir.files:
+                    writer.writerow(
+                        {
+                            "path": file.path,
+                            "source_chars": (
+                                len(file.templated_file.source_str)
+                                if file.templated_file
+                                else ""
```

Comment:
Nitpick: Since these output fields are all integers, would `0` make more sense than `""` as the `else` value?

Reply:
My intent here was to output something more like a `null` value than a 0. It didn't take zero time, it didn't happen at all. I think zeros could mess up averages further down the line.